src/execution/sql/cte_scan_iterator.cpp

```diff
@@ -0,0 +1,69 @@
+#include "execution/sql/cte_scan_iterator.h"
+#include "parser/expression/constant_value_expression.h"
+#include "transaction/deferred_action_manager.h"
+#include "transaction/transaction_context.h"
+
+namespace terrier::execution::sql {
+
+parser::ConstantValueExpression DummyCVE() {
+  return parser::ConstantValueExpression(type::TransientValueFactory::GetInteger(0));
+}
+
+CteScanIterator::CteScanIterator(terrier::execution::exec::ExecutionContext *exec_ctx, uint32_t *schema_cols_type,
+                                 uint32_t num_schema_cols)
+    : exec_ctx_(exec_ctx), cte_table_oid_(static_cast<catalog::table_oid_t>(999)), table_redo_(nullptr) {
```

Comment:
Magic number? Can you make this a macro?

Reply:
How do you know 999 won't conflict with any other table_oid? Will that cause issues?